backend/src/main/java/pl/zzpj/esportbetting/impl/UserDetailsServiceImpl.java

```diff
@@ -120,4 +121,31 @@ public User getUser(Authentication principal) {
         UserDetailsImpl userDetails = (UserDetailsImpl) principal.getPrincipal();
         return UserDetailsImplToUserConverter.convert(userDetails);
     }
+
+    @Override
+    public Statistics getUserStats(User user) {
+        List<Bet> userFinishedBets = user.getBets().stream()
+                                            .filter(b -> b.getMatch().getStatus() == MatchStatusEnum.FINISHED)
+                                            .collect(Collectors.toList());
+
+        int goodBets = 0;
+        int badBets = 0;
+        int earnedCoins = 0;
+        int lostCoins = 0;
+
+        for (Bet bet: userFinishedBets) {
+            DetailedFinishedStatusEnum winnerTeam = bet.getMatch().getWhichTeamWon();
+            if ((winnerTeam.equals(DetailedFinishedStatusEnum.A_WIN) && bet.isSelectedA()) ||
+                    (winnerTeam.equals(DetailedFinishedStatusEnum.B_WIN) && bet.isSelectedA() && !bet.isSelectedA())) {
```

Comment:
Czy po "||"  nie powinno być: " (winnerTeam.equals(DetailedFinishedStatusEnum.B_WIN) && !bet.isSelectedA())" ??

Reply:
To jest magia kopiuj-wklej xD